### PR TITLE
Minor updates to multi-gpu training

### PR DESCRIPTION
**Pull request by anwai98**
Hi @constantinpape,

I made some minor changes to the DDP-based training to fit our SAM finetuning. ~There are a few issues in handling `mixed_precision`, would be good to take a look at it.~

### CODE REVIEW

**Comment by anwai98** [comment]
For reference, this is the error I get:

`AssertionError: No inf checks were recorded for this optimizer.`

**Comment by anwai98** [comment]
@constantinpape This is GTG from my side.

File: torch_em/multi_gpu_training.py [hunk 1]
```diff
@@ -33,37 +33,82 @@ def _create_data_loader(ds_callable, ds_kwargs, loader_kwargs, world_size, rank)
     return loader
 
 
+class DDP(DistributedDataParallel):
```
Comment (constantinpape) [inline torch_em/multi_gpu_training.py:36]: Can you explain why this is necessary?

File: torch_em/multi_gpu_training.py [hunk 2]
```diff
@@ -33,37 +33,82 @@ def _create_data_loader(ds_callable, ds_kwargs, loader_kwargs, world_size, rank)
...
-    loader_kwargs, iterations, **kwargs
+    model_callable,
+    model_kwargs,
+    train_dataset_callable,
+    train_dataset_kwargs,
+    val_dataset_callable,
+    val_dataset_kwargs,
+    loader_kwargs,
+    iterations,
+    find_unused_parameters=False,
```
Comment (constantinpape) [inline torch_em/multi_gpu_training.py:105]: From what I understand this is important to make the training work for more complex models. So should we set the default to `True`?

**Review by constantinpape — COMMENTED** [review]
Looks good overall, only two minor comments.

File: torch_em/multi_gpu_training.py (hunk 1 above)
Comment (anwai98) [reply torch_em/multi_gpu_training.py:36]: Ah yes, to access the original model's functionalities (say `self.model.image_embeddings_oft` for instance), the calls from `DDP` wrapped models are `self.model.module.image_embeddings_oft`. I decided to generalize this via another class to automate this, rather changing it on the model-level calls.

File: torch_em/multi_gpu_training.py (hunk 2 above)
Comment (anwai98) [reply torch_em/multi_gpu_training.py]: Yep, you are correct. I've set the default value to True now. Thanks for the suggestion!

File: torch_em/multi_gpu_training.py (hunk 1 above)
Comment (constantinpape) [reply torch_em/multi_gpu_training.py:36]: I see! That makes sense. This is a good design! Can you add a short comment on this?

Comment (anwai98) [reply torch_em/multi_gpu_training.py:36]: Thanks! Added a docstring to the wrapper now!

**Review by constantinpape — APPROVED** [review]

**Pull request merged**